pkg/api/mock/utils_test.go

```diff
@@ -0,0 +1,101 @@
+package mock
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Mock utils", func() {
+	Context("merge", func() {
+		var (
+			res interface{}
+			err error
+		)
+
+		It("can merge primitives", func() {
+			type test struct {
+				String string `json:"string,omitempty"`
+				Int    int    `json:"int,omitempty"`
+			}
+
+			res, err = merge(&test{}, &test{String: "test", Int: 1})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res).To(Equal(&test{String: "test", Int: 1}))
+
+			res, err = merge(&test{Int: 1}, &test{String: "test"})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res).To(Equal(&test{String: "test", Int: 1}))
+
+			res, err = merge(&test{String: "test", Int: 1}, &test{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res).To(Equal(&test{String: "test", Int: 1}))
+
+			res, err = merge(&test{String: "test", Int: 1}, &test{String: "test_changed", Int: 2})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res).To(Equal(&test{String: "test_changed", Int: 2}))
```

Comment:
this check is now practically the same first one, the second Object overriding all properties - change it to only updating one field and checking the other field is still the same after merge

Reply:
I would have updated the string attribute, but that's not important ^^